tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java

```diff
@@ -2626,9 +2626,19 @@ private String handleDateFormatItem(String xpath, String value, boolean showCont
                     }
                 } else {
                     List<String> examples = new ArrayList<>();
-                    examples.add(sdf.format(DATE_SAMPLE3));
-                    examples.add(sdf.format(DATE_SAMPLE));
-                    examples.add(sdf.format(DATE_SAMPLE4));
+                    DayPeriodInfo dayPeriodInfo = supplementalDataInfo.getDayPeriods(DayPeriodInfo.Type.format, cldrFile.getLocaleID());
+                    Set<DayPeriod> dayPeriods = new LinkedHashSet<DayPeriod>(dayPeriodInfo.getPeriods());
+                    for (DayPeriod dayPeriod : dayPeriods) {
+                        if (dayPeriod.equals(DayPeriod.midnight)) { // suppress midnight, see ICU-12278 bug
+                            continue;
+                        }
+                        R3<Integer, Integer, Boolean> info = dayPeriodInfo.getFirstDayPeriodInfo(dayPeriod);
+                        if (info != null) {
+                            int time = ((info.get0() + info.get1()) / 2); // time midpoint 
```

Comment:
Why the midpoint? I'm cool with it, I'm just not sure why then midpoint was chosen.

Reply:
You're right, the midpoint specifically is an arbitrary choice--it could be any value in between the start/end time. Setting the sample to either the start time (`info.get0()`) or the end time (`info.get1()`) doesn't predictably get the intended day period, since day periods have overlapping start/end times (ex. noon starts/ends at 12pm and afternoon starts at 12pm, so using the start time as the sample produces two examples formatted as 12:00 noon, vs two distinct examples). So, I thought the midpoint would be the simplest in-between value :).

Reply:
I'll modify the comment to include that the midpoint was chosen to disambiguate!